### PR TITLE
Adding several distributions.

### PR DESCRIPTION
**Pull request by Sahand-H**
# Distributions:
- Logistic
- LogLogistic
- F
- Weibull
- Gumbel
- Gompertz

### CODE REVIEW

File: .settings/ca.ubc.stat.blang.BlangDsl.prefs [hunk 1]
```diff
@@ -1,15 +1,16 @@
-//outlet.DEFAULT_OUTPUT.sourceFolder.src/main/java.directory=build/blang/main
-//outlet.DEFAULT_OUTPUT.sourceFolder.src/main/resources.directory=build/blang/main
-//outlet.DEFAULT_OUTPUT.sourceFolder.src/test/java.directory=build/blang/test
+#Thu Nov 29 11:03:07 PST 2018
```
Comment (alexandrebouchard) [inline .settings/ca.ubc.stat.blang.BlangDsl.prefs]: Please avoid committing .prefs files

File: src/main/java/blang/distributions/F.bl [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package blang.distributions
+
+/** The F-distribution. Also known as Fisher-Snedecor distribution. */
```
Comment (alexandrebouchard) [inline src/main/java/blang/distributions/F.bl]: Give the support of the distribution.

File: src/main/java/blang/distributions/F.bl [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  laws {
+	logf(d1, d2) {
+		if (d1 <= 0.0) return NEGATIVE_INFINITY
+		if (d2 <= 0.0) return NEGATIVE_INFINITY
+		return 0.5 * (d1*log(d1) + d2*log(d2))
+	}
+	logf(d1, d2) {
+		if (d1 <= 0.0) return NEGATIVE_INFINITY
+		if (d2 <= 0.0) return NEGATIVE_INFINITY
+		return lnGamma(d1/2) + lnGamma(d2/2) - lnGamma((d1 + d2)/2)
```
Comment (alexandrebouchard) [inline src/main/java/blang/distributions/F.bl]: Better to combine logf blocks with identical dependencies (i.e. the first two here)

File: src/main/java/blang/distributions/F.bl [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	logf(d1, d2, realization) {
+		if (d1 <= 0.0) return NEGATIVE_INFINITY
+		if (d2 <= 0.0) return NEGATIVE_INFINITY
+		if ((d1 == 1) && (realization <= 0.0)) return NEGATIVE_INFINITY
+		if (realization < 0.0) return NEGATIVE_INFINITY
+		return - 0.5 * (d1 + d2) * log( (d1 * realization) + d2) 
+	}
+  }
+	
+  generate(rand) {
```
Comment (alexandrebouchard) [inline src/main/java/blang/distributions/F.bl]: Indentation inconsistent. Preferably use 2-space based indentation.

File: src/main/java/blang/distributions/Generators.java [hunk 5]
```diff
@@ -25,6 +28,64 @@
 /** Various random number generators. */
 public class Generators // Warning: blang.distributions.Generators hard-coded in ca.ubc.stat.blang.scoping.BlangImplicitlyImportedFeatures 
 {
+  /** */
+  public static double gompertz(Random rand, double shape, double scale)
+  {
+    double percentile = rand.nextDouble();
+    //double result = - (Math.log(shape) - Math.log(shape - Math.log(1 - percentile)*Math.log(scale))) / Math.log(scale);
```
Comment (alexandrebouchard) [inline src/main/java/blang/distributions/Generators.java]: Avoid commented lines of code in main branch.

File: src/main/java/blang/distributions/Generators.java [hunk 6]
```diff
@@ -25,6 +28,64 @@
...
+	if (result == 0.0)
+		result = ZERO_PLUS_EPS;
+	return result;
+  }
+  
+  /** */
+  public static double logisticDist(Random rand, double location, double scale) 
+  {
+	double result = new LogisticDistribution(generator(rand), location, scale).sample();
+	if (result == 0.0)
```
Comment (alexandrebouchard) [inline src/main/java/blang/distributions/Generators.java]: The logistic distribution has positive density at zero, so please remove this if here

File: src/main/java/blang/distributions/Generators.java [hunk 7]
```diff
@@ -25,6 +28,64 @@
...
+		result = ZERO_PLUS_EPS;
+	return result;
+  }
+  
+  /** */
+  public static double logLogistic(Random rand, double scale, double shape)
```
Comment (alexandrebouchard) [inline src/main/java/blang/distributions/Generators.java]: Same as last comment here.

File: src/main/java/blang/distributions/Gompertz.bl [hunk 8]
```diff
@@ -0,0 +1,34 @@
+package blang.distributions
+
+/** The Gompertz distribution. Values in \([0, \infty) \).
```
Comment (alexandrebouchard) [inline src/main/java/blang/distributions/Gompertz.bl]: Minidoc comments should be all in one line

File: .settings/ca.ubc.stat.blang.BlangDsl.prefs (hunk 1 above)
Comment (Sahand-H) [reply .settings/ca.ubc.stat.blang.BlangDsl.prefs]: Sorry about that!

**Comment by Sahand-H** [comment]
Made the requested changes, hopefully nothing else slipped through the cracks!

**Comment by alexandrebouchard** [comment]
Many thanks!

**Pull request merged**